rust/src/dataset.rs

```diff
@@ -499,24 +503,36 @@ impl Dataset {
         self.append_impl(batches, params).await
     }
 
+    async fn latest_manifest(&self) -> Result<Manifest> {
+        read_manifest(&self.object_store, &latest_manifest_path(&self.base)).await
+    }
+
     /// Restore the currently checked out version of the dataset as the latest version.
     ///
     /// Currently, `write_params` is just used to get additional store params.
     /// Other options are ignored.
     pub async fn restore(&mut self, write_params: Option<WriteParams>) -> Result<()> {
-        // TODO: once we merge conflict resolution, we should be able to remove
-        // this and just rely on the commit process.
-        let new_version = self.latest_manifest().await?.version + 1;
```

Comment:
If you call restore while on the latest version, should the correct behavior be that a new version is created or should it just be a noop?

Reply:
It should probably be a no-op but the code right now will create a new version.